packages/@ember/-internals/glimmer/lib/resolver.ts

```diff
@@ -55,24 +54,31 @@ function instrumentationPayload(name: string) {
   return { object: `component:${name}` };
 }
 
-function componentFor(name: string, owner: Owner): Option<Factory<object> | object> {
-  let fullName = `component:${name}`;
+function componentFor(
+  name: string,
+  owner: InternalOwner
+): Option<InternalFactory<object> | object> {
+  let fullName = `component:${name}` as const;
```

Comment:
What's the point of `as const` here?

Reply:
Throughout, they all tell TS to notice the actual structure of the string, which in turn makes any manually-constructed string like this valid to use as a `FullName` as long as it has the right shape.

Reply:
Ah, got it!